src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -30,6 +30,8 @@ protected MessageBody(Frame context)
 
         public bool RequestUpgrade { get; protected set; }
 
+        public virtual bool IsEmpty => false;
+
         public Task<int> ReadAsync(ArraySegment<byte> buffer, CancellationToken cancellationToken = default(CancellationToken))
         {
             var task = PeekAsync(cancellationToken);
```

Comment:
After discussing in person, I like returning a ReadableBuffer and appending that to the RequestBodyReader pipe even more. That way we can even keep the CopyToAsyncDoesNotCopyBlocks test 😄